lib/widgets/gridcards.dart

```diff
@@ -0,0 +1,334 @@
+import 'package:flutter/material.dart';
+import 'package:google_fonts/google_fonts.dart';
+import 'package:sochem/utils/constants.dart';
+
+class HomeScreenGrid extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            Padding(
+              padding:
+                  const EdgeInsets.only(top: 20.0, left: 15.0, right: 15.0),
+              child: Container(
+                height: MediaQuery.of(context).size.height * 0.15,
+                width: MediaQuery.of(context).size.width * 0.40,
+                decoration: BoxDecoration(
+                  borderRadius: BorderRadius.circular(25),
+                  color: cardImageColor,
+                  boxShadow: [
+                    BoxShadow(
+                      color: Colors.black45,
+                      offset: const Offset(
+                        1.0,
+                        1.0,
+                      ), //Offset
+                      blurRadius: 5.0,
+                    ),
+                  ],
+                ),
+                child: Column(
+                  children: [
+                    Padding(
+                      padding: const EdgeInsets.only(top: 10.0),
+                      child: Center(
+                        child: Image.asset(
+                          'assets/feed.png',
+                          width: 60.0,
+                          height: 60.0,
+                        ),
+                      ),
+                    ),
+                    Padding(
+                      padding: const EdgeInsets.only(top: 5.0),
+                      child: Text(
+                        "Feed",
+                        style: GoogleFonts.montserrat(
+                          textStyle: TextStyle(
+                            fontSize: 20,
+
+                            fontWeight: FontWeight.w600,
+                            color: Colors.black,
+                            // letterSpacing: 1.2,
+                          ),
+                        ),
+                      ),
+                    ),
+                  ],
+                ),
+              ),
+            ),
+            Padding(
+              padding:
+                  const EdgeInsets.only(top: 20.0, left: 15.0, right: 15.0),
+              child: Container(
+                height: MediaQuery.of(context).size.height * 0.15,
```

Comment:
@ItsAdityaKSingh I see a repetitive use of widgets for making individual cards. You can extract out a single widget take two parameters about the image path and text to be displayed instead of repeating it manually 6 times.

Also isn't there a native way to make grid layout in flutter? You are manually making columns and rows to achieve the purpose which is not preffered.

Reply:
There were UI problems while using GridView as individual cards were adapting their heights according to the width of the column. So I had to do it this way. I started with GridView but was required to go with columns.
And about the cards, sure I'll do the needful.

Reply:
Acknowledged